Encore Version Patch
=======================

This patch modifies the Go build tools (those which build Go itself) to add a "encore" prefix to the version tag, this
allows us to identify if the Go runtime being used is an Encore patched runtime or it is a standard Go runtime.

It changes the output of `go version` from:
    `go version go1.17.6 darwin/arm64`
to:
    `go version encore-go1.17.6 darwin/arm64`

diff --git a/src/cmd/dist/build.go b/src/cmd/dist/build.go
index c36a12e5e9..ee9910401c 100644
--- a/src/cmd/dist/build.go
+++ b/src/cmd/dist/build.go
@@ -362,7 +362,7 @@ func findgoversion() string {
 					b = "builder " + hostType
 				}
 			}
-			return b
+			return strings.Replace(b, "go1.", "encore-go1.", 1)
 		}
 	}

@@ -393,7 +393,7 @@ func findgoversion() string {
 	if m == nil {
 		fatalf("internal/goversion/goversion.go does not contain 'const Version = ...'")
 	}
-	version := fmt.Sprintf("devel go1.%s-", m[1])
+   version := fmt.Sprintf("devel encore-go1.%s-", m[1])
 	version += chomp(run(goroot, CheckExit, "git", "log", "-n", "1", "--format=format:%h %cd", "HEAD"))

 	// Cache version.
